userspace/libsinsp/events/sinsp_events_ppm_sc.cpp

```diff
@@ -312,3 +320,70 @@ std::unordered_set<std::string> libsinsp::events::sc_set_to_names(const libsinsp
 	}
 	return ppm_sc_names_set;
 }
+
+libsinsp::events::set<ppm_sc_code> libsinsp::events::sinsp_repair_state_sc_set(const libsinsp::events::set<ppm_sc_code>& ppm_sc_set)
+{
+	uint32_t flags = 0;
+	if (!libsinsp::events::net_sc_set().intersect(ppm_sc_set).empty())
+	{
+		flags |= PPM_ADAPTIVE_SC_NETWORK_BASE;
+		flags |= PPM_ADAPTIVE_SC_FD_CLOSE;
+	}
+
+	static libsinsp::events::set<ppm_sc_code> accept_listen_sc_set = {PPM_SC_ACCEPT, PPM_SC_ACCEPT4, PPM_SC_LISTEN};
+	if (!accept_listen_sc_set.intersect(ppm_sc_set).empty())
+	{
+		flags |= PPM_ADAPTIVE_SC_NETWORK_BASE;
+		flags |= PPM_ADAPTIVE_SC_NETWORK_BIND;
+		flags |= PPM_ADAPTIVE_SC_FD_CLOSE;
+	}
+
+	if (!libsinsp::events::file_sc_set().intersect(ppm_sc_set).empty() ||
+		!libsinsp::events::io_sc_set().intersect(ppm_sc_set).empty() ||
+		!libsinsp::events::io_other_sc_set().intersect(ppm_sc_set).empty())
+	{
+		flags |= PPM_ADAPTIVE_SC_FD_CLOSE;
+	}
+
+	/* These syscalls are used to build up or modify info of the basic process (tinfo) struct.
+	 * Consistent enforcement regardless of the input ppm_sc_set.
+	 */
+	libsinsp::events::set<ppm_sc_code> repaired_sinsp_state_sc_set = {
```

Comment:
Yes! Nice, since now we can do it, left a comment stating that it is ok if this enforcement is redundant in case we decide to also enforce `PPM_SC_SCHED_PROCESS_EXIT` elsewhere.

@jasondellaluce when we refactor `sc_names_to_sc_set` and/or create `event_names_to_sc_set` need to also include the tp now ...